truffle/src/main/java/org/jruby/truffle/parser/BodyTranslator.java

```diff
@@ -1826,14 +1826,17 @@ public RubyNode visitHashNode(HashParseNode node) {
         final SourceSection fullSourceSection = sourceSection.toSourceSection(source);
 
         final List<RubyNode> hashConcats = new ArrayList<>();
+        final List<Boolean> hashConcatsIsMerge = new ArrayList<>();
 
         final List<RubyNode> keyValues = new ArrayList<>();
 
         for (KeyValuePair<ParseNode, ParseNode> pair: node.getPairs()) {
             if (pair.getKey() == null) {
```

Comment:
This case with a null key means that we have a ** splat like in `{a: 1, **{b: 2}, c: 3}`.
It would merit a comment for the least!